test/test_torch.py

```diff
@@ -2244,6 +2280,31 @@ def run_test(A_dims, b_dims, cast):
     def test_lu_solve_batched_many_batches(self):
         self._test_lu_solve_batched_many_batches(self, lambda t: t)
 
+    @staticmethod
+    def _test_lu_solve_batched_dims(self, cast, pivot=True):
+        from numpy.linalg import solve
+        from common_utils import random_fullrank_matrix_distinct_singular_value
+
+        def run_test(A_dims, b_dims, cast, pivot):
+            A = random_fullrank_matrix_distinct_singular_value(*A_dims)
+            b = torch.randn(*b_dims)
+            x_exp = torch.as_tensor(solve(A.numpy(), b.numpy()))
+            A, b = cast(A), cast(b)
+            LU_data, LU_pivots = torch.lu(A, pivot=pivot)
+            x = torch.lu_solve(b, LU_data, LU_pivots)
+            self.assertEqual(x, cast(x_exp))
+
+        # test against numpy.linalg.solve
+        run_test((4, 2, 1, 3), (2, 1, 3, 4, 6), cast, pivot)  # no broadcasting
```

Comment:
(4, 2, 3, 1) is really hard to read. It's saying that `A` is a tensor with size (2, 3, 1, 4, 4), but it takes a bit of flipping through the code to figure this out.

A few action items here:
- `run_test`'s first argument shouldn't be `A_dims`... it should probably be `A_batch_dims` and then `A_last_two_dims` or something.
- random_fullrank_matrix_distinct_singular_value has two arguments, `l` and `*batches`. We should probably rename them to be a little more understandable, i.e. "last_two_dims" and "batch_dims".

Reply:
What I'll do is: fix this line to be
```
run_test((2, 1, 3, 4, 4), (2, 1 3, 4, 6), cast, pivot)
```
and reconstruct the tuple to be passed to `random_fullrank_matrix_distinct_singular_value` and then rename the arguments as suggested.